src/operator/mxnet_op.h

```diff
@@ -241,20 +243,26 @@ struct Kernel;
 template<typename OP>
 struct Kernel<OP, cpu> {
   template<typename ...Args>
-  inline static void Launch(mshadow::Stream<cpu> *s, int N, Args... args) {
-#if (MXNET_USE_CUDA == 0)
-    #pragma omp parallel for
-#endif
-    for (int i = 0; i < N; ++i) {
-      OP::Map(i, args...);
+  inline static void Launch(mshadow::Stream<cpu> *s, const int N, Args... args) {
+    const int omp_cores = Engine::Get()->GetNumOMPThreadsPerWorker();
+    if (omp_cores <= 1) {
+      // Zero means not to use OMP, but don't interfere with external OMP behavior
+      for (int i = 0; i < N; ++i) {
+        OP::Map(i, args...);
+      }
+    } else {
+      #pragma omp parallel for num_threads(omp_cores - 1)
```

Comment:
why -1?

Reply:
I am changing to omp_threads only. Was leaving a thread for the engine. However in this PR I only wish to keep same OMP behavior other than consistent behavior between CPU and GPU builds.

Reply:
Leaving 1 for the engine.
Eventually probably OMP will be, for example, divided by number of concurrent ops running. Also I am working on tuning which will likely come into play as well. For a later PR early next week.